lib/web_ui/dev/safari.dart

```diff
@@ -43,21 +35,24 @@ class Safari extends Browser {
         infoLog: DevNull(),
       );
 
-      // Safari will only open files (not general URLs) via the command-line
-      // API, so we create a dummy file to redirect it to the page we actually
-      // want it to load.
-      final Directory redirectDir = Directory(
-        path.join(environment.webUiDartToolDir.path),
-      );
-      final redirect = path.join(redirectDir.path, 'redirect.html');
-      File(redirect).writeAsStringSync(
-          '<script>location = ' + jsonEncode(url.toString()) + '</script>');
-
-      var process =
-          await Process.start(installation.executable, [redirect] /* args */);
-
-      unawaited(process.exitCode
-          .then((_) => File(redirect).deleteSync(recursive: true)));
+      // In the latest versions of MacOs opening Safari browser with a file brings
+      // a popup which halts the test.
+      // The following list of arguments needs to be provided to the `open` command
+      // to open Safari for a given URL. In summary they provide a new instance
+      // to open, that instance to wait for opening the url until Safari launches,
+      // provide Safari bundles identifier.
+      // For more details run `man open` on MacOS.
+      // TODO(nurhan): explore implementing this part using Apple Script or
```

Comment:
Consider creating an issue and linking to it from here.

Reply:
done.